personal_profile/tests.py

```diff
@@ -68,3 +64,26 @@ def test_profile_is_deleted_when_user_is_deleted(self, profile_1, user_1):
         test_user = User.objects.filter(username='user_1').first()
         test_user.delete()
         assert test_profile not in PersonalProfile.objects.all()
+
+
+@pytest.mark.django_db
+class TestProfileDetailView:
+    def test_detail_view_page_entrypoint(self, profile_1, user_1, client):
+        # Testing to see if a valid user id gets a valid detail view page
+        response = client.get(PROFILE_DETAIL_URL + str(profile_1.id) + '/')
```

Comment:
How the user logged in? 
You want to make sure only the user can edit his profile?
consider the client option for login
see https://pytest-django.readthedocs.io/en/latest/helpers.html#id2

force login example

Reply:
I added the login of a user to the tests although for now, we thought to minimize the personal profile to only display the details with no option to update according to what Barak has said on the last demo (to not waste time on "create, update and delete" if it's not unique)- though if we have time I will add it.
That means that every user can look at other users' profiles.

Reply:
So is the feature not important at all?
If you propose a PR is should be properly tested,
If not consider if PR is required at all

Reply:
This PR is for the display of personal profile details.
On a different PR, I will add the update option and then the relevant tests accordingly (maybe it will happen after another more important PR).

Reply:
But there is a way to login the user
Please avoid this 
> test_user = User.objects.filter(username='user_1').first()
Please  refer samples for force_login

Reply:
@ShirleyFichman ^^

Reply:
Actually there is no need for a user to be logged in at all, so you could drop the client login at all

Reply:
I am not sure If it is seen from the test,
There is an assert for the context returned from the client

Reply:
> Actually there is no need for a user to be logged in at all, so you could drop the client login at all

Prefered to have full path coverage for gui test insted of the View tests.

views tests without clients does not cover links errors and returned pages.

Reply:
I meant that this view doesn't require a logged in user so the `client.login` or `client.force_login` is unnecessary.
I think that lines 78 and 79 should be dropped and the remaining test should remain as it is, only the context returned assertion should be changed to:
`assert response.context['personalprofile'] == user_1.personalprofile`.
because the existing `response.context['user']` only refers to the logged in user and not any object that is returned from `views.py`.

Reply:
@paOmer , you are right, thanks for noticing- I will change accordingly.